build.go

```diff
@@ -54,17 +54,55 @@ func Build(procs ProcMgr, logger scribe.Emitter) packit.BuildFunc {
 			return packit.BuildResult{}, errors.New("need exactly one of: $PHP_HTTPD_PATH or $PHP_NGINX_PATH")
 		}
 
+		shouldEnableReload, err := reloader.ShouldEnableLiveReload()
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
 		logger.Process("Determining start commands to include in procs.yml:")
 		// HTTPD Case
 		if httpdConfPath != "" {
 			serverProc := NewProc("httpd", []string{"-f", httpdConfPath, "-k", "start", "-DFOREGROUND"})
+
+			if exists, err := fs.Exists(filepath.Join(context.WorkingDir, ".httpd.conf.d")); err != nil {
+				return packit.BuildResult{}, err
+			} else if shouldEnableReload && exists {
+				serverProc = NewProc("watchexec", []string{
+					"--watch", "/workspace/.httpd.conf.d",
+					"--on-busy-update", "signal",
+					"--signal", "SIGHUP",
+					"--", "httpd",
+					"-f", httpdConfPath, "-k",
+					"start",
+					"-DFOREGROUND",
```

Comment:
* `--on-busy-update=signal` means issue a signal to the process that watchexec owns when something watched has changed
* `--signal=SIGHUP` means to issue the hangup/HUP specifically

Normally we use `--restart` since we do want the entire process restarted in most cases. The three processes we care about (`nginx`, `httpd`, and `fpm`) all respond to a `HUP` signal by reloading configuration.

* `nginx`: http://nginx.org/en/docs/control.html
* `httpd`: https://httpd.apache.org/docs/2.4/stopping.html
* `fpm`: reference TBD

```shell
% watchexec --help
watchexec 1.20.4
Execute commands when watched files change

USAGE:
    watchexec [OPTIONS] <command>...

...

Behaviour options:
        --on-busy-update <on-busy-update>    Select the behaviour to use when receiving events while the command is running. Current default is queue, will change to do-nothing in 2.0. [possible
                                             values: do-nothing, queue, restart, signal]
    -r, --restart                            Restart the process if it's still running. Shorthand for --on-busy-update=restart
    -s, --signal <signal>                    Specify the signal to send when using --on-busy-update=signal
```

Reply:
I have to run an errand. When I get back I will update this PR to fix the integration test and update since `main` was fixed, and respond to the various comments.

Reply:
Hmmm.... I can't seem to find a reference that FPM will reload configuration when it receives the `HUP` signal. It does seem to react to the `USR2` signal though. https://linux.die.net/man/8/php-fpm

This open issue from 2014 would indicate that FPM does not react to the `HUP` signal. https://bugs.php.net/bug.php?id=67553.

I will see what manual testing I can do.

Reply:
Ah great work discovering that FPM catch. I'll take a look as well

Reply:
Alright testing FPM (the web-server reload on config changes work flawlessly :D):
- Built the HTTPD reload test app with the reloadable process env var set.
- Created a `user.conf` file with absurd configuration:
```
; User-set FPM configuration
[www]
; The number of child processes to be created when pm is set to 'static' and the
; maximum number of child processes when pm is set to 'dynamic' or 'ondemand'.

pm.max_children = 2000
```
- copied it into the running container under `/workspace/.php.fpm.d/user.conf`
- I saw no logs from the container about changes to FPM configuration
- I was still able to curl the container as normal
- I deleted the `user.conf` file from `.php.fpm.d/`
- I saw logs from the container about FPM failing to initialize:
```
[07-Oct-2022 19:21:07] WARNING: Nothing matches the include pattern '/workspace/.php.fpm.bp/*.conf' from /layers/paketo-buildpacks_php-fpm/php-fpm-config/base.conf at line 30.
[07-Oct-2022 19:21:07] WARNING: Nothing matches the include pattern '/workspace/.php.fpm.d/*.conf' from /layers/paketo-buildpacks_php-fpm/php-fpm-config/base.conf at line 32.
[07-Oct-2022 19:21:07] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[07-Oct-2022 19:21:07] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[07-Oct-2022 19:21:07] ERROR: unable to bind listening socket for address '127.0.0.1:9000': Address already in use (98)
[07-Oct-2022 19:21:07] ERROR: FPM initialization failed
[[Command exited with 78]]
```
- I still see an FPM process running on the container (the original one i think)
- I can still curl the container endpoint no problem

Reply:
For test 2:
- I modified `  "--signal", "SIGHUP",` in the FPM block to be ` "--signal", "SIGUSR2",` per your recommendation and rebuilt the buildpack
- Did the same steps as before
- Copying the `user.conf` file in caused the container to only spit out:
```
[[Command killed by User2]]
```
- It appears that FPM is still running on the container and I can access the container endpoint
- VERY interestingly, if I copy in the file again (that is already on the container file system) I get:
```
[07-Oct-2022 19:43:25] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[07-Oct-2022 19:43:25] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[07-Oct-2022 19:43:25] ERROR: unable to bind listening socket for address '127.0.0.1:9000': Address already in use (98)
[07-Oct-2022 19:43:25] ERROR: FPM initialization failed
[[Command exited with 78]]
```
- The original warning that there's nothing inside of `.php.fpm.d` is gone, so FPM has realized we've added something
- It can't restart FPM it seems because the binding socket is still intact

Reply:
So, essentially with both signal types issued in the watchexec process, neither one is able to kill the connection fully and reload

Reply:
Need to do some more investigation

Reply:
> So, essentially with both signal types issued in the watchexec process, neither one is able to kill the connection fully and reload

What was the issue here? Not going to block on this since the tests pass & I see that they check for successful reload of the procs but just wondering what the resolution was, if any? Did you observe this as well, @joshuatcasey?

Reply:
@sophiewigmore and I paired to check this, we were able to get the expected behavior in that FPM reloads configuration when it receives the `SIGUSR2` signal.